src/figuresClasses.ts

```diff
@@ -1,19 +1,73 @@
-export interface Figure {
+type Shape = 'triangle' | 'rectangle' | 'circle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some((argument: number) => argument <= 0)) {
+      throw new Error('Sides can not be less than 0');
+    }
 
-export class Circle {
+    if (a >= b + c
+      || b >= a + c
+      || c >= a + b) {
+      throw new Error('Is not triangle');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor((Math.sqrt(4 * this.a ** 2 * this.b ** 2
+    - (this.a ** 2 + this.b ** 2 - this.c ** 2) ** 2)) * 0.25 * 100) / 100;
+  }
```

Comment:
There was another way of Heron's formula, that way was easier for me. It is only my opinion))